src/formatDate.js

```diff
@@ -50,7 +50,60 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let year = '';
+  let month = '';
+  let day = '';
+  const [,,, separatorFF] = fromFormat;
+  const [,,, separatorNF] = toFormat;
+
+  const dates = date.split(separatorFF);
+  const newFormat = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YY':
+      case 'YYYY':
+        year = dates[i];
+        break;
+      case 'MM':
+        month = dates[i];
+        break;
+      case 'DD':
+        day = dates[i];
+        break;
```

Comment:
The switch/case construction should always have a default case to avoid unpredictable issues.